backend/src/org/commcare/suite/model/RemoteQueryDatum.java

```diff
@@ -26,10 +28,10 @@
     public RemoteQueryDatum() {
     }
 
-    public RemoteQueryDatum(String url, String storageInstance,
+    public RemoteQueryDatum(URL url, String storageInstance,
                             Hashtable<String, XPathExpression> hiddenQueryValues,
                             Hashtable<String, DisplayUnit> userQueryPrompts) {
-        super(storageInstance, url);
+        super(storageInstance, url.toString());
```

Comment:
This is pretty messy, but I didn't really see a better way given constraints. 

Here is my sorta hand wavy justification:
I'm utilizing the `value` field of the `SessionDatum` super class to store the url. But since we have a custom serialization framework, we can't just serialize the URL, we need to turn it into a String first.

Given that the `url` field is externally used as a `URL`, I figured we should represent it as one coming in as well, so that the only place that it isn't a `URL` is inside the class abstraction.